app/Jobs/SchedulePostJob.php

```diff
@@ -0,0 +1,41 @@
+<?php
+
+namespace App\Jobs;
+
+use Carbon\Carbon;
+use Illuminate\Bus\Queueable;
+use Illuminate\Contracts\Queue\ShouldBeUnique;
+use Illuminate\Contracts\Queue\ShouldQueue;
+use Illuminate\Foundation\Bus\Dispatchable;
+use Illuminate\Queue\InteractsWithQueue;
+use Illuminate\Queue\SerializesModels;
+
+class SchedulePostJob implements ShouldQueue
+{
+    use Dispatchable, InteractsWithQueue, Queueable, SerializesModels;
+
+    /**
+     * Create a new job instance.
+     *
+     * @return void
+     */
+    public function __construct()
+    {
+        //
+    }
+
+    /**
+     * Execute the job.
+     *
+     * @return void
+     */
+    public function handle()
+    {
+        $date = Carbon::now();
+        $currenTtimestamp = $date->toDateTimeString();
+        $posts = \App\Models\Post::where('post_schedule_time', '>', $currenTtimestamp)->get();
+        foreach ($posts as $post) {
```

Comment:
running queries in loop is worst. You can simply run update query with where.

Reply:
resolved